chain/cmd/bandoracled2/handler.go

```diff
@@ -1,126 +1,97 @@
-	if len(dataSourceIDs) != len(calldataList) {
-		logger.Error("😱 Request #%d: inconsistent data source count and calldata count", id)
+		l.Error(":cold_sweat: Failed to convert %s to integer with error: %s", idStr, err.Error())
 		return
 	}
 
-	reports := make([]oracle.RawReport, 0)
-	// TODO: Parallelize the work to get data from each source.
-	for idx := range dataSourceIDs {
-		dataSourceID, err := strconv.Atoi(dataSourceIDs[idx])
-		if err != nil {
-			logger.Error(
-				"❌ Request #%d: failed to parse data source id %s with error: %d",
-				id, dataSourceIDs[idx], err,
-			)
-			return
-		}
-
-		externalID, err := strconv.Atoi(externalIDs[idx])
-		if err != nil {
-			logger.Error(
-				"❌ Request #%d: failed to parse external id %s with error: %d",
-				id, externalIDs[idx], err,
-			)
-			return
-		}
+	l = l.With("rid", id)
+	l.Info(":delivery_truck: Processing incoming request event")
 
-		executable, err := GetExecutable(client, dataSourceID)
-		if err != nil {
-			logger.Error(
-				"❌ Request #%d: failed to get executable of data source id #%d with error: %d",
-				id, dataSourceID, err,
-			)
-			return
-		}
+	// TODO: Skip if not related to this validator
+	reqs, err := GetRawRequests(log)
+	if err != nil {
+		l.Error(":skull: Failed to parse raw requests with error: %s", err.Error())
+	}
 
-		// TODO: Parallelize the work. Remove hardcode.
-		result, err := byteexec.RunOnAWSLambda(
-			executable, 3*time.Second, calldataList[idx],
-			"https://dmptasv4j8.execute-api.ap-southeast-1.amazonaws.com/bash-execute",
-		)
-		// TODO: Extract exit code.
-		if err != nil {
-			logger.Error(
-				"❌ Request #%d: failed to run executable of data source id #%d with error: %d",
-				id, dataSourceID, err,
+	reportsChan := make(chan oracle.RawReport, len(reqs))
+	for _, req := range reqs {
+		go func(l *Logger, req rawRequest) {
+			exec, err := GetExecutable(c, l, int(req.dataSourceID))
+			if err != nil {
+				l.Error(":skull: Failed to load data source with error: %s", err.Error())
+				reportsChan <- oracle.NewRawReport(
+					req.externalID, 255, []byte("FAIL_TO_LOAD_DATA_SOURCE"),
+				)
+				return
+			}
+			executor := &lambdaExecutor{}
+			result, exitCode := executor.Execute(l, exec, 3*time.Second, req.calldata)
+			l.Debug(
+				":sparkles: Query data done with calldata: %q, result: %q, exitCode: %d",
+				req.calldata, result, exitCode,
 			)
-			return
-		}
-		reports = append(reports, oracle.NewRawReport(oracle.ExternalID(externalID), 0, result))
+			reportsChan <- oracle.NewRawReport(req.externalID, exitCode, result)
+		}(l.With("did", req.dataSourceID, "eid", req.externalID), req)
 	}
 
-	sort.Slice(reports, func(i, j int) bool {
-		return reports[i].ExternalID < reports[j].ExternalID
-	})
+	reports := make([]oracle.RawReport, 0)
+	for _ = range reqs {
+		reports = append(reports, <-reportsChan)
+	}
 
-	BroadCastMsgs(client, key, []sdk.Msg{
+	// TODO: Parallelize this and make sure we don't do multiple broadcasts at the same time.
```

Comment:
I think `Parallelize this` will make confusion later. Just left Make sure sequence number correct when send tx at the same time